### PR TITLE
Analysis move range

### PR DESCRIPTION
**Pull request by jacobm-tech**
Initial implementation of restricting additional game analysis to a move range. Work in progress. Several Kivy details to be worked out, i18n not done. Issue #562.

### CODE REVIEW

File: katrain/gui/popups.py [hunk 1]
```diff
@@ -839,9 +841,25 @@ def on_submit(self):
...
+        self.checkbox.bind(active=self.on_checkbox_active)
+
+        self.start_move = find_child_by_name(self, "start_move")
+        self.end_move = find_child_by_name(self, "end_move")
+        self.start_move.disabled = True
+        self.end_move.disabled = True
+        self.start_move.text = str(katrain.game.current_node.depth)
+
     def on_submit(self):
         self.button.trigger_action(duration=0)
```
Comment (sanderland) [inline katrain/gui/popups.py:864]: please check the kv for how 'button' is a property here, so you can do the same with other fields

**Comment by sanderland** [comment]
I think we can simplify this by just having empty inputs go to 0 and inf defaults and not have a checkbox.

**Comment by jacobm-tech** [comment]
I really like the idea of filling the starting move with the current move number, which I think would be one of the most common uses for it, and this requires a checkbox. If it gets too complicated, I can get rid of it, but I don't think the checkbox itself is the problem, the i18n work is the big expense here, and you need most of it anyway.

File: katrain/gui/popups.py (hunk 1 above)
Comment (jacobm-tech) [reply katrain/gui/popups.py:864]: Oh, I see, I did not understand what was going on with these properties. I think I'll be able to get rid of the get child by name stuff.

Comment (jacobm-tech) [reply katrain/gui/popups.py:859]: It's gone.

**Comment by jacobm-tech** [comment]
OK, I tried a few things, but I can't get the dialog to look the way I want. I would like something like

```
+-------------------------------------+
|                                     |
|                         +--+        |
|   Visits per move:      |  |        |
|                         +--+        |
|                                     |
|  +-+                    +--+  +--+  |
|  | | Limit to moves:    |  |  |  |  |
|  +-+                    +--+  +--+  |
|                                     |
|                                     |
|  +-+                                |
|  | | Mistakes only                  |
|  +-+                                |
|                                     |
|         +-----------------+         |
|         |   Reanalyze     |         |
|         +-----------------+         |
|                                     |
+-------------------------------------+
```

with the checkmarks aligned and all the text the same size. Instead, I get 3 different alignments and 4 different text sizes no matter what I do. I think I need some help with this. Functionally, it works the way I wanted.

I will start the i18n work.

**Comment by sanderland** [comment]
have a look at 70113c0

And yes, kivy is pain

**Comment by jacobm-tech** [comment]
Version with i18n. Strings done for Russian.

**Comment by sanderland** [comment]
Great. Ready for merge?

File: katrain/core/game.py [hunk 2]
```diff
@@ -601,6 +601,10 @@ def analyze_extra(self, mode, **kwargs):
         if mode == "game":
             nodes = self.root.nodes_in_tree
             only_mistakes = kwargs.get("mistakes_only", False)
+            move_range = kwargs.get("move_range", False)
+            if move_range:
```
Comment (sanderland) [inline katrain/core/game.py:605]: one thought: maybe this could be 
move_range is either absent/None or (start,end)

Comment (jacobm-tech) [reply katrain/core/game.py:605]: Not sure what you mean - would the call in popups.kv be like this?
```
root.katrain("analyze-extra", "game", visits=visits.input_value, move_range=(start_move.input_value, end_move.input_value) if move_range.input_value else None, mistakes_only=mistakes.input_value)
```

Comment (sanderland) [reply katrain/core/game.py:605]: yep

Comment (jacobm-tech) [reply katrain/core/game.py:605]: I changed it to that style.

**Comment by jacobm-tech** [comment]
Ready for merge, but still needs i18n strings for everything except Russian and French.

**Comment by sanderland** [comment]
> Ready for merge, but still needs i18n strings for everything except Russian and French.

No worries about that. I tend to let them pile up a bit and then contact people at the next 'bigger' release.

**Pull request merged**